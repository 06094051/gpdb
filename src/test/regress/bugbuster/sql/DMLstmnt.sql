CREATE DOMAIN country_code char(2) NOT NULL;
ALTER DOMAIN country_code DROP NOT NULL;



DROP DOMAIN country_code;

CREATE FUNCTION addlab1111(integer,integer) RETURNS integer
AS 'select $1 + $2;'
LANGUAGE SQL 
IMMUTABLE
CONTAINS SQL
RETURNS NULL ON NULL INPUT;

ALTER FUNCTION addlab1111(integer,integer) RENAME TO addition111;


DROP FUNCTION addition111(integer,integer);

CREATE USER jonathan11 WITH PASSWORD 'abc1';
CREATE USER jonathan12 WITH PASSWORD 'abc2';

ALTER USER jonathan11 RENAME TO jona11;
ALTER USER jonathan12 RENAME TO jona12;

DROP USER jona11;
DROP USER jona12;

CREATE USER jonathan12 WITH PASSWORD 'abc2';

ALTER USER jonathan11 RENAME TO jona11;
ALTER USER jonathan12 RENAME TO jona12;

CREATE GROUP marketing WITH USER jona11,jona12;

ALTER GROUP marketing RENAME TO market;

DROP GROUP market;
DROP USER jona11;
DROP USER jona12;

select unnest('{}'::text[]);

drop table mpp_r6756 cascade; --ignore
drop table mpp_s6756 cascade; --ignore

create table mpp_r6756 ( a int, b int, x int, y int ) 
    distributed randomly
    partition by list(a) (
        values (0), 
        values (1)
        );

create table mpp_s6756 ( c int, d int, e int ) 
    distributed randomly;

insert into mpp_s6756 values 
    (0,0,0),(0,0,1),(0,1,0),(0,1,1),(1,0,0),(1,0,1),(1,1,0),(1,1,1);

insert into mpp_r6756 values
    (0, 0, 1, 1),
    (0, 1, 2, 2),
    (0, 1, 2, 2),
    (1, 0, 3, 3),
    (1, 0, 3, 3),
    (1, 0, 3, 3),
    (1, 1, 4, 4),
    (1, 1, 4, 4),
    (1, 1, 4, 4),
    (1, 1, 4, 4);

-- start_equiv
select a, b, count(distinct x), count(distinct y)
from mpp_r6756 r, mpp_s6756 c, mpp_s6756 d, mpp_s6756 e
where r.a = c.c and r.a = d.d and r.a = e.e
group by a, b
union all
select a, null, count(distinct x), count(distinct y)
from mpp_r6756 r, mpp_s6756 c, mpp_s6756 d, mpp_s6756 e
where r.a = c.c and r.a = d.d and r.a = e.e
group by a
union all
select null, null, count(distinct x), count(distinct y)
from mpp_r6756 r, mpp_s6756 c, mpp_s6756 d, mpp_s6756 e
where r.a = c.c and r.a = d.d and r.a = e.e;

select a, b, count(distinct x), count(distinct y)
from mpp_r6756 r, mpp_s6756 c, mpp_s6756 d, mpp_s6756 e
where r.a = c.c and r.a = d.d and r.a = e.e
group by rollup (a,b);
--end_equiv

drop table mpp_r6756 cascade; --ignore
drop table mpp_s6756 cascade; --ignore

--start_ignore
drop table if exists copytest;
--end_ignore
