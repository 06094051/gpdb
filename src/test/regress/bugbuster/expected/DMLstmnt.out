CREATE DOMAIN country_code char(2) NOT NULL;
ALTER DOMAIN country_code DROP NOT NULL;
DROP DOMAIN country_code;
CREATE FUNCTION addlab1111(integer,integer) RETURNS integer
AS 'select $1 + $2;'
LANGUAGE SQL
IMMUTABLE
CONTAINS SQL
RETURNS NULL ON NULL INPUT;
ALTER FUNCTION addlab1111(integer,integer) RENAME TO addition111;
DROP FUNCTION addition111(integer,integer);
CREATE USER jonathan11 WITH PASSWORD 'abc1';
NOTICE:  resource queue required -- using default resource queue "pg_default"
CREATE USER jonathan12 WITH PASSWORD 'abc2';
NOTICE:  resource queue required -- using default resource queue "pg_default"
ALTER USER jonathan11 RENAME TO jona11;
NOTICE:  MD5 password cleared because of role rename
ALTER USER jonathan12 RENAME TO jona12;
NOTICE:  MD5 password cleared because of role rename
DROP USER jona11;
DROP USER jona12;
CREATE USER jonathan12 WITH PASSWORD 'abc2';
NOTICE:  resource queue required -- using default resource queue "pg_default"
ALTER USER jonathan11 RENAME TO jona11;
ERROR:  role "jonathan11" does not exist
ALTER USER jonathan12 RENAME TO jona12;
NOTICE:  MD5 password cleared because of role rename
CREATE GROUP marketing WITH USER jona11,jona12;
NOTICE:  resource queue required -- using default resource queue "pg_default"
ERROR:  role "jona11" does not exist
ALTER GROUP marketing RENAME TO market;
ERROR:  role "marketing" does not exist
DROP GROUP market;
ERROR:  role "market" does not exist
DROP USER jona11;
ERROR:  role "jona11" does not exist
DROP USER jona12;
select unnest('{}'::text[]);
 unnest 
--------
(0 rows)

drop table mpp_r6756 cascade; --ignore
ERROR:  table "mpp_r6756" does not exist
drop table mpp_s6756 cascade; --ignore
ERROR:  table "mpp_s6756" does not exist
create table mpp_r6756 ( a int, b int, x int, y int ) 
    distributed randomly
    partition by list(a) (
        values (0), 
        values (1)
        );
NOTICE:  CREATE TABLE will create partition "mpp_r6756_1_prt_1" for table "mpp_r6756"
NOTICE:  CREATE TABLE will create partition "mpp_r6756_1_prt_2" for table "mpp_r6756"
create table mpp_s6756 ( c int, d int, e int ) 
    distributed randomly;
insert into mpp_s6756 values 
    (0,0,0),(0,0,1),(0,1,0),(0,1,1),(1,0,0),(1,0,1),(1,1,0),(1,1,1);
insert into mpp_r6756 values
    (0, 0, 1, 1),
    (0, 1, 2, 2),
    (0, 1, 2, 2),
    (1, 0, 3, 3),
    (1, 0, 3, 3),
    (1, 0, 3, 3),
    (1, 1, 4, 4),
    (1, 1, 4, 4),
    (1, 1, 4, 4),
    (1, 1, 4, 4);
-- start_equiv
select a, b, count(distinct x), count(distinct y)
from mpp_r6756 r, mpp_s6756 c, mpp_s6756 d, mpp_s6756 e
where r.a = c.c and r.a = d.d and r.a = e.e
group by a, b
union all
select a, null, count(distinct x), count(distinct y)
from mpp_r6756 r, mpp_s6756 c, mpp_s6756 d, mpp_s6756 e
where r.a = c.c and r.a = d.d and r.a = e.e
group by a
union all
select null, null, count(distinct x), count(distinct y)
from mpp_r6756 r, mpp_s6756 c, mpp_s6756 d, mpp_s6756 e
where r.a = c.c and r.a = d.d and r.a = e.e;
 a | b | count | count 
---+---+-------+-------
   |   |     4 |     4
 0 | 0 |     1 |     1
 0 | 1 |     1 |     1
 1 | 0 |     1 |     1
 1 | 1 |     1 |     1
 0 |   |     2 |     2
 1 |   |     2 |     2
(7 rows)

select a, b, count(distinct x), count(distinct y)
from mpp_r6756 r, mpp_s6756 c, mpp_s6756 d, mpp_s6756 e
where r.a = c.c and r.a = d.d and r.a = e.e
group by rollup (a,b);
 a | b | count | count 
---+---+-------+-------
 0 | 0 |     1 |     1
 0 | 1 |     1 |     1
 1 | 0 |     1 |     1
 1 | 1 |     1 |     1
 0 |   |     2 |     2
 1 |   |     2 |     2
   |   |     4 |     4
(7 rows)

--end_equiv
drop table mpp_r6756 cascade; --ignore
drop table mpp_s6756 cascade; --ignore
--start_ignore
drop table if exists copytest;
--end_ignore
